src/modules/musig/musig.md

```diff
@@ -0,0 +1,196 @@
+MuSig - Rogue-Key-Resistant Multisignatures Module
+===========================
+
+This module implements the MuSig [1] multisignature scheme. The majority of
+the module is an API designed to be used by signing or auditing participants
+in a multisignature scheme. This involves a somewhat complex state machine
+and significant effort has been taken to prevent accidental misuse of the
+API in ways that could lead to accidental signatures or loss of key material.
+
+# Theory
+
+In MuSig all participants contribute key material to a single signing key,
+using the equation
+
+    P = sum_i µ_i - P_i
+
+where `P_i` is the public key of the `i`th participant and `µ_i` is a so-called
+_MuSig coefficient_ computed according to the following equation
+
+    L = H(P_1 || P_2 || ... || P_n)
+    µ_i = H(L || i)
+
+where H is a hash function modelled as a random oracle.
+
+To produce a multisignature `(s, R)` on a message `m` using verification key
+`P`, signers act as follows:
+
+1. Each computes a nonce, or ephemeral keypair, `(k_i, R_i)`. Every participant
+   communinicates `H(R_i)` to every other participant.
+2. Upon receipt of every `H(R_i)`, each participant communicates `R_i` to every
+   other participant. The recipients check that each `R_i` is consistent with
+   the previously-communicated hash.
+3. Each participant computes a combined nonce
+    `R = sum_i R_i`
+   and shared challenge
+    `e = H(R || P || m)`
+   and partial signature
+    `s_i = k_i + µ_i*x_i*e`
+   where `x_i` is the secret key corresponding to `P_i`.
+
+The complete signature is then the `(s, R)` where `s = sum_i s_i` and `R = sum_i R_i`.
+
+# API Usage
+
+The following sections describe use of our API, and are mirrored in code in `src/modules/musig/example.c`.
+
+It is essential to security that signers use a unique uniformly random nonce for all
+signing sessions, and that they do not reuse these nonces even in the case that a
+signing session fails to complete. To that end, all signing state is encapsulated
```

Comment:
well they should not reuse it, no matter if the signing sessions fails to complete or  not. ;)

Reply:
I think the word "even" communicates this.

Reply:
agreed